src/pages/main/components/selected-dataset/build-flow/components/build-flow-header.tsx

```diff
@@ -46,15 +46,17 @@ export const BuildFlowHeader = observer(
     return (
       <div className={styles.buildFlow__header}>
         <div className="flex items-center">
-          <div
-            onClick={handleGoBack}
-            className={styles.buildFlow__header__button}
-          >
-            <Icon
-              name="Arrow"
-              className={styles.buildFlow__header__button__icon}
-            />
-          </div>
+          {!isXL && wizardStore.actionHistory.historyIndex <= 1 ? null : (
```

Comment:
It's not so easy to read.
I'd suggest to create new variable like `isBackButtonShown` for this conditions and to use it without null: `isBackButtonShown && (<div ...`